### DISCUSSION TITLE
unstable_cache, revalidateTag and when revalidation occurs

### DISCUSSION THREAD

**Discussion by derekgygax** [vercel/next.js#63120] [Help]
### Summary

My project runs full stack using nextjs. The project displays and edits events that are stored in a postgres database. I have two pages, the home page at '/' and a page for editing an event at '/edit-event'. Both pages make a database call to get the list of events, the function uses unstable_cache and has the tag 'events_list'.

```javascript
const getEventsList = unstable_cache(
    getEventsListFunction, 
    ['initiatives_list'], 
    { 
         tags: ['events_list']
     });
```
The homepage lists the events in a table and the edit-event page gives you a drop down to select an event by name which will retrieve the event details. There is also a page to add new events '/add-event'. When you add an event a server action is called that does 

`revalidateTag('events_list')`

When the project is built on Vercel everything is statically generated. There are no dynamic pages.

When you add an event and revalidateTag('events_list') is called I was under the impression that this would lead to the home page and the dropdown in the add events page to be revalidated and contain the new event added to the database. This however is not the case in the Vercel build. However, if I do a production build on my local machine it is working how I understood it would. Also worth nothing that I navigate between the pages using the Link component in next. If I manually force a reload on the home page the new event will show up. However, even when I do this forced reload on the edit event page, the new event does not show up in the drop down.

Am I misunderstanding things or is this not behaving as intended?

**Comment by z0n** [comment]
I have a similar question and I don't think I understand how `unstable_cache()` works.

My setup:
- I'm hosting Next.js on Vercel.
- I have a `/listing/[...slug]` `page.tsx` where I generate paths at build time using `generateStaticParams()` (the pages are something like product pages).
- Inside that page.tsx, I fetch data for the page on server-side using a third-party library ([contentful](https://www.npmjs.com/package/contentful)).
  - To improve page speed, I wrapped `unstable_cache()` around the contentful fetch call with options: `{ revalidate: 60 * 60 * 24, tags: ['listing', slug] }`.
  - I also have `export const revalidate = 60 * 60 * 24;` inside page.tsx to be able to update listings without re-deploying.

Now what I'm expecting:
- When I open the page after 24 hours, the data for the page is fetched (and cached) again.
- When I cause a manual `revalidateTag('listing')` (I have a special API route based on [this](https://nextjs.org/docs/app/api-reference/functions/revalidatePath#route-handler)), the page is fetching new data the next time I open it.

I can't find a way to reliably validate if this is working. I can't find anything in the Vercel logs regarding revalidation, even when I trigger it manually via my revalidation route.

Can anyone confirm if it works like I expect or am I missing something? I can't share the code unfortunately but I may be able to add more insights if it helps.

**Reply by flnx** [reply]
Did you find a solution? I'm facing the same issue. The unstable_cache() wouldn't revalidate at all.

**Reply by z0n** [reply]
Nope, I'm afraid not.
